Automattic/woocommerce-payments
# Remove Order Meta Helper tool toggle, and add url parameter to enable tool

Fixes #6935 #6919

#### Changes proposed in this Pull Request

* Remove the feature flag for the Order Meta Helper.
* Remove the functionality to add and use the button to toggle the helper on and off.
* Add `try/catch` for call to `get_intent()` to handle if the intent is invalid.
* Remove `$is_wcpay` variable/check since it's not needed anymore.
* Add functionality to enable the tool on a per-order basis via a url parameter: `&wcpay_mc_meta_helper=1`

Note that the `maybe_update_edit_post_link` method seems redundant that it checks to see if the feature is enabled in order to update the url/link. This is needed for when the form is submitted and the page is refreshed. The `$_REQUEST` variable is checked and the `hidden` input is used to make sure the url is updated so the form shows again.

#### Testing instructions

Setup:
* Go to WooCommerce > Settings > Multi-Currency and enable at least one other currency in your store.

Testing:
* Place an order for a product using the default store currency.
* View the order under WooCommerce > Orders in the admin.
* Confirm there is not a meta box named _Multi-Currency Meta Helper_ showing.
* Update the url in the address bar to have `&wcpay_mc_meta_helper=1` on the end and hit return/enter.
* The meta box should still not show.
* Purchase a product in another currency other than the default currency in your store.
* Go to the order under WooCommerce > Orders in the admin.
* Confirm there is not a meta box named _Multi-Currency Meta Helper_ showing.
* Update the url in the address bar to have `&wcpay_mc_meta_helper=1` on the end and hit return/enter.
* Confirm that the meta box named _Multi-Currency Meta Helper_ now shows.
* Copy and paste the suggested rate into the box and hit enter/return or click the _Update_ button for the order.
* A new note should show in the order notes under your user ID.
* The note should state the exchange rate was updated and include the from and to rates.
* The _Multi-Currency Exchange Rate_ value should now be populated in the meta box, the _Stripe Exchange Rate_ value will not be.
* Update the payment intent ID on the order to either remove or add a character on the end.
  * This can be done by adding this snippet somewhere, which enables viewing hidden meta data: `add_filter( 'is_protected_meta', '__return_false', 999 );`, or
  * This can be done in the `wp_postmeta` table by searching for your order number under `post_id`.
* Refresh the order.
* The meta box should now show:
  *  _Charge Exchange Rate_ is _Not found_.
  *  _Charge Currency_ is _Not found_. 
* Check the latest WCPay log either under WooCommerce > Status > Logs or /wp-content/uploads/wc-logs/ for a log that reads something like the below example.
* Go to Products in the left navigation bar, then into an existing product.
* Update the url in the address bar to have `&wcpay_mc_meta_helper=1` on the end and hit return/enter.
* Confirm there is not a meta box named _Multi-Currency Meta Helper_ showing.

Example log:
> 2023-08-04T15:24:42+00:00 ERROR Error when attempting to get intent (pi_3NayaBR8d1QhXXSD0Pwx3yHJ_):
Error: No such payment_intent: 'pi_3NayaBR8d1QhXXSD0Pwx3yHJ_'


-------------------


**Post merge**
